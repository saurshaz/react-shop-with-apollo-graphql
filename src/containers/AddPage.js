import React from "react";
import { Link } from "react-router-dom";
import { graphql} from "react-apollo";
import gql from "graphql-tag";
import {
  Button,
  Grid,
  Typography,
  TextField,
} from "@material-ui/core";

// TODO: rm hardcoding, take values from form only for image URLs
const ADD_PRODUCT = gql`mutation newOrderItem($description: String!, $miniDescription: String!, $discount: Int!, $price: Int!) {
  createOrderItem(
    description: $description,
    miniDescription: $miniDescription,
    discount: $discount,
    price: $price,
    images:{
      thumb: "https://lorempixel.com/400/200/sports/shop-me-here/",
      small:"https://lorempixel.com/600/400/sports/shop-me-here/",
      large:"https://lorempixel.com/1200/800/sports/shop-me-here/"
    }
  ) {
    id
    description
    miniDescription
    images {
      thumb
      small
      large
    }
  }
}`;


class AddPage extends React.Component {

  constructor(props){
    super(props)
    this.state = {}
  }

  addOrderItem(){
    const { description, miniDescription, price, discount } = this.state;
    this.props
      .mutate({
        variables: { description, miniDescription, price, discount }
      })
      .then(data => {
        console.log("product added");
      })
      .catch(err => {
        console.log(err);
      });
  }

  render(){
    return (
      <div>
        <Link
          to="/"
          className="fixed bg-white top-0 right-0 pa4 ttu dim black no-underline"
        >
          <Button variant="contained" color="primary">
            Home
          </Button>
        </Link>
        <React.Fragment>
          <Typography variant="h6" gutterBottom>
            Add a Product
          </Typography>
          <Grid container spacing={24}>
            <Grid item xs={12} sm={6}>
              <TextField
                required
                id="description"
                name="description"
                label="Description"
                fullWidth
                autoComplete="description"
                onChange={e => this.setState({ description: e.target.value })}
              />
            </Grid>
            <Grid item xs={12} sm={6}>
              <TextField
                required
                id="miniDescription"
                name="miniDescription"
                label="mini Description"
                fullWidth
                autoComplete="miniDescription"
                onChange={e => this.setState({ miniDescription: e.target.value })}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                required
                id="price"
                name="price"
                label="Price"
                fullWidth
                autoComplete="price"
                onChange={e => this.setState({ price: parseInt(e.target.value || 0, 10) })}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                id="discount"
                name="discount"
                label="Discount(in percentage)"
                fullWidth
                autoComplete="discount"
                onChange={e => this.setState({ discount: parseInt(e.target.value || 0, 10) })}
              />
            </Grid>
            <Grid item xs={12}>
              <Button
                variant="contained"
                color="primary"
                onClick={this.addOrderItem.bind(this)}
              >
                Add Product
              </Button>
            </Grid>
          </Grid>
        </React.Fragment>
      </div>
    );
  }
};

export default graphql(ADD_PRODUCT)(AddPage);
// export default AddPage;
