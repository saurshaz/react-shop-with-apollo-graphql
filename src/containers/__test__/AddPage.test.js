// Link.react.test.js
import React from 'react';
import AddPage from '../AddPage';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() })

test('AddPage component test', () => {

    const wrapper = shallow(
        <AddPage mutate={ () => {}} />
    );
    expect(wrapper).toMatchSnapshot();
});