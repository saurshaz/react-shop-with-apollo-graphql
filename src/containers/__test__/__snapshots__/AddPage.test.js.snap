// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPage component test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Apollo(AddPage)
    mutate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "ignoreResults": true,
      "mutation": Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "newOrderItem",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "description",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "description",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "miniDescription",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "miniDescription",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "discount",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "discount",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "price",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "price",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "images",
                      },
                      "value": Object {
                        "fields": Array [
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "thumb",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "https://lorempixel.com/400/200/sports/shop-me-here/",
                            },
                          },
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "small",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "https://lorempixel.com/600/400/sports/shop-me-here/",
                            },
                          },
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "large",
                            },
                            "value": Object {
                              "block": false,
                              "kind": "StringValue",
                              "value": "https://lorempixel.com/1200/800/sports/shop-me-here/",
                            },
                          },
                        ],
                        "kind": "ObjectValue",
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "createOrderItem",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "description",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "miniDescription",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "images",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "thumb",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "small",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "large",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "description",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "miniDescription",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Int",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "discount",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "Int",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "price",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 568,
          "start": 0,
        },
      },
      "variables": undefined,
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "ignoreResults": true,
        "mutation": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "newOrderItem",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "description",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "description",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "miniDescription",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "miniDescription",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "discount",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "discount",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "price",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "price",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "images",
                        },
                        "value": Object {
                          "fields": Array [
                            Object {
                              "kind": "ObjectField",
                              "name": Object {
                                "kind": "Name",
                                "value": "thumb",
                              },
                              "value": Object {
                                "block": false,
                                "kind": "StringValue",
                                "value": "https://lorempixel.com/400/200/sports/shop-me-here/",
                              },
                            },
                            Object {
                              "kind": "ObjectField",
                              "name": Object {
                                "kind": "Name",
                                "value": "small",
                              },
                              "value": Object {
                                "block": false,
                                "kind": "StringValue",
                                "value": "https://lorempixel.com/600/400/sports/shop-me-here/",
                              },
                            },
                            Object {
                              "kind": "ObjectField",
                              "name": Object {
                                "kind": "Name",
                                "value": "large",
                              },
                              "value": Object {
                                "block": false,
                                "kind": "StringValue",
                                "value": "https://lorempixel.com/1200/800/sports/shop-me-here/",
                              },
                            },
                          ],
                          "kind": "ObjectValue",
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "createOrderItem",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "description",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "miniDescription",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "images",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "thumb",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "small",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "large",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "description",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "miniDescription",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "Int",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "discount",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "Int",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "price",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 568,
            "start": 0,
          },
        },
        "variables": undefined,
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
